snow/consensus/snowball/network_test.go

```diff
@@ -26,34 +28,44 @@ func (n *Network) Initialize(params Parameters, numColors int) {
 	}
 }
 
-func (n *Network) AddNode(sb Consensus) {
+func (n *Network) AddNode(newConsensusFunc newConsensusFunc) Consensus {
+	consensus := newConsensusFunc(n.params, n.colors[0])
+
 	s := sampler.NewUniform()
 	s.Initialize(uint64(len(n.colors)))
 	indices, _ := s.Sample(len(n.colors))
-	sb.Initialize(n.params, n.colors[int(indices[0])])
 	for _, index := range indices[1:] {
-		sb.Add(n.colors[int(index)])
+		consensus.Add(n.colors[int(index)])
 	}
```

Comment:
I know this doesn't change any functionality... But this looks wrong to me. By slicing `indices` aren't we going to potentially not register a color?